src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +375,61 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
 		}
 		ioOperations.deleteInventory(medicalInventory);
 	}
+
+	/**
+	 * Confirm the Inventory rows of inventory.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
+	 * @throws OHDataValidationException
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public List<Movement> confirmMedicalInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		//check if invenotry exist
+		int id = inventory.getId();
+		inventory = this.getInventoryById(id);
+		// validate the inventory
+		this.validateMedicalInventoryRow(inventory, inventoryRowSearchList);
+		// get general info
+		String referenceNumber = inventory.getInventoryReference();
+		// TODO: make possibility to allow charges and discharges with same referenceNumber
+		String chargeReferenceNumber = referenceNumber + "-charge";
+		String dischargeReferenceNumber = referenceNumber + "-discharge";
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(inventory.getChargeType());
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(inventory.getDischargeType());
+		Supplier supplier = supplierManager.getByID(inventory.getSupplier());
+		Ward ward = wardManager.findWard(inventory.getDestination());
+		LocalDateTime now = TimeTools.getNow();
+		// prepare movements
+		List<Movement> chargeMovements = new ArrayList<>();
+		List<Movement> dischargeMovements = new ArrayList<>();
+		for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+			MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+
+			double theoQty = medicalInventoryRow.getTheoreticQty();
+			double realQty = medicalInventoryRow.getRealQty();
+			Double ajustQty = realQty - theoQty;
+			Medical medical = medicalInventoryRow.getMedical();
+			Lot currentLot = medicalInventoryRow.getLot();
+			if (ajustQty > 0) { // charge movement when realQty > theoQty
+				Movement movement = new Movement(medical, chargeType, null, currentLot, now, ajustQty.intValue(), supplier, chargeReferenceNumber);
+				chargeMovements.add(movement);
+			} else if (ajustQty < 0) { // discharge movement when realQty < theoQty
+				Movement movement = new Movement(medical, dischargeType, ward, currentLot, now, -(ajustQty.intValue()), null, dischargeReferenceNumber);
```

Comment:
```suggestion
				Movement movement = new Movement(medical, dischargeType, ward, currentLot, now, -ajustQty.intValue(), null, dischargeReferenceNumber);
```
Unnecessary parentheses